### ISSUE TITLE
"react/jsx-wrap-multilines": "ternaries" option

### ISSUE THREAD

**Issue by zorzysty** [yannickcr/eslint-plugin-react#1114]
I propose to add an option to [react/jsx-wrap-multilines rule](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-wrap-multilines.md) that would make this code valid:
```
const header = this.props.header
  ? <HeaderElement
      className="myclass myclass2"
      isExpanded={this.state.isExpanded}
  >
    {this.props.header}
  </HeaderElement>
  : null;
```
There is no real gain from adding parentheses here. It doesn't improve readability, but adds extra noise:
```
const header = this.props.header
  ? (
    <HeaderElement
      className="myclass myclass2"
      isExpanded={this.state.isExpanded}
    >
      {this.props.header}
    </HeaderElement>
  )
  : null;
```
or even worse:
```
const header = this.props.header
  ? (<AccordionElement
    className="myclass myclass2"
    isExpanded={this.state.isExpanded}
  >
    {this.props.header}
  </AccordionElement>)
  : null;
```
Right now to make omitting parentheses viable I need to switch off both "assignment" and "declaration" options, but because of this I'm loosing errors for code like this:
```
const header =
  <div>
    xyz
  </div>;
```
which I believe is much better with parentheses: 
```
const header = (
  <div>
    xyz
  </div>
);
```

**Comment by ljharb** [comment]
This is how I'd indent your code:
```jsx
const header = this.props.header ? (
  <HeaderElement
      className="myclass myclass2"
      isExpanded={this.state.isExpanded}
  >
    {this.props.header}
  </HeaderElement>
 ) : null;
```

Does that example help?

**Comment by zorzysty** [comment]
Yeah, it's better, but it is still [conflicting with prettier](https://prettier.github.io/prettier/#%7B%22content%22%3A%22const%20header%20%3D%20this.props.header%20%3F%20(%5Cn%20%20%3CHeaderElement%5Cn%20%20%20%20%20%20className%3D%5C%22myclass%20myclass2%5C%22%5Cn%20%20%20%20%20%20isExpanded%3D%7Bthis.state.isExpanded%7D%5Cn%20%20%3E%5Cn%20%20%20%20%7Bthis.props.header%7D%5Cn%20%20%3C%2FHeaderElement%3E%5Cn%20)%20%3A%20null%3B%22%2C%22options%22%3A%7B%22printWidth%22%3A80%2C%22tabWidth%22%3A2%2C%22singleQuote%22%3Afalse%2C%22trailingComma%22%3A%22none%22%2C%22bracketSpacing%22%3Atrue%2C%22jsxBracketSameLine%22%3Afalse%2C%22parser%22%3A%22babylon%22%2C%22doc%22%3Afalse%7D%7D), which opts for this kind of formatting for ternaries:
```javascript
const header = this.props.header
  ? <ElementRenederedWhenTrue />
  : <ElementRenederedWhenFalse />;
```
And i'm just trying to make Airbnb ESlint rules work with prettier. Here's analogical issue I reported for prettier: https://github.com/prettier/prettier/issues/1009#issuecomment-286833002

**Comment by ljharb** [comment]
Airbnb won't be changing any of our eslint rules to match prettier; all our rules are the way they are for a reason.

The latter example your posted is just fine, because the jsx is single-line. Airbnb's style guide (and this rule) requires parens around multiline ternaries only.

**Comment by zorzysty** [comment]
Yes, I get that Airbnb won't change the rules, but I can override them so they will work with prettier.
Unfortunately to override this particular one I need to switch off both "assignment" and "declaration", which is a shame. If there was an option to switch off requiring parens just for multiline ternaries a lot more cases would be covered.
